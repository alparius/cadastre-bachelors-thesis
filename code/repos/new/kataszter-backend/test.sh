#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
