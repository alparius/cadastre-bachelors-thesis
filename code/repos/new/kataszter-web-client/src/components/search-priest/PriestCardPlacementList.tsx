import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { NavLink } from "react-router-dom";
import { List } from "semantic-ui-react";

import IPlacement from "../../data/priest/Placement";
import ItalicFM from "../../util/styledMessages/ItalicFM";
import DateDisplay from "../DateDisplay";

interface IPriestCardPlacementListProps {
    places: IPlacement[];
    headerKey: string;
}

const PriestCardPlacementList: React.FC<IPriestCardPlacementListProps> = (props: IPriestCardPlacementListProps) => {
    const { places, headerKey } = props;

    return (
        <List>
            <List.Item>
                <List.Header>
                    <FM id={headerKey} />
                </List.Header>
            </List.Item>
            {places ? (
                places.map((place: IPlacement, index: number) => (
                    <List.Item key={place.place + index}>
                        {place.place ? (
                            <NavLink to={`/parishes/profile/${place.placeID}`} className="hoverLink">
                                {" "}
                                {place.place}{" "}
                            </NavLink>
                        ) : (
                            <ItalicFM id="general.noData" />
                        )}
                        : <DateDisplay date={place.startDate} /> - <DateDisplay date={place.endDate} />
                    </List.Item>
                ))
            ) : (
                <List.Item>
                    <ItalicFM id="general.noData" />
                </List.Item>
            )}
        </List>
    );
};

export default PriestCardPlacementList;
