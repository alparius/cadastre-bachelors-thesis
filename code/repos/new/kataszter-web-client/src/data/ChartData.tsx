interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
