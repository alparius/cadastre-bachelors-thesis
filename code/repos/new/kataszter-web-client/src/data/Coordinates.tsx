interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
