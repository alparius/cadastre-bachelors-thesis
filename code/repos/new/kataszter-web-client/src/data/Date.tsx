interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
