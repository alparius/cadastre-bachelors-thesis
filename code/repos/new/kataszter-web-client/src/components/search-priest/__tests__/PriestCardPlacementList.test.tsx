/*===================
  PriestCardPlacementList.unit.test.tsx
 ===================*/
import React from "react";

import IPlacement from "../../../data/priest/Placement";
import { NewTestDate, NewTestPlacement } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import PriestCardPlacementList from "../PriestCardPlacementList";

describe("PriestCardPlacementList element", () => {
    it("should render a priest placement list", () => {
        const regularPlacements: IPlacement[] = [NewTestPlacement];
        const container = createComponentWithIntlAndRouter(<PriestCardPlacementList places={regularPlacements} headerKey="test" />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a priest placement list with 'no data' labels for missing fields", () => {
        const placementWithMissingFields: IPlacement = { place: "place", startDate: NewTestDate, endDate: NewTestDate } as any;
        const placementsWithMissingFields: IPlacement[] = [placementWithMissingFields];
        const container = createComponentWithIntlAndRouter(
            <PriestCardPlacementList places={placementsWithMissingFields} headerKey="test" />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an empty list", () => {
        const emptyPlacements: IPlacement[] = [];
        const container = createComponentWithIntlAndRouter(<PriestCardPlacementList places={emptyPlacements} headerKey="test" />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
