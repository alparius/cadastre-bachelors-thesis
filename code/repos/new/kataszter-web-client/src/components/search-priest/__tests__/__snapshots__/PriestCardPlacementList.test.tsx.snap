// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestCardPlacementList element should render a priest placement list 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      test
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <a
      aria-current={null}
      className="hoverLink"
      href="/parishes/profile/0"
      onClick={[Function]}
    >
       
      place
       
    </a>
    : 
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      2000.06.06
    </span>
     - 
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      2000.06.06
    </span>
  </div>
</div>
`;

exports[`PriestCardPlacementList element should render a priest placement list with 'no data' labels for missing fields 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      test
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <a
      aria-current={null}
      className="hoverLink"
      href="/parishes/profile/undefined"
      onClick={[Function]}
    >
       
      place
       
    </a>
    : 
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      2000.06.06
    </span>
     - 
    <span
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      2000.06.06
    </span>
  </div>
</div>
`;

exports[`PriestCardPlacementList element should render an empty list 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      test
    </div>
  </div>
</div>
`;
